test/ConstValues/SectionIR.swift

```diff
@@ -28,8 +28,14 @@ func bar(x: Int) -> String { return "test" }
 @section("mysection") let funcRef1 = foo // ok
 @section("mysection") let funcRef2 = bar // ok
 
-// metatypes - TODO
-//@section("mysection") let metatype1 = Int.self
+struct S: Hashable, Sendable {}
+
+// metatypes
+@section("mysection") let metatype1 = Int.self
+@section("mysection") let metatype2: Any.Type = Int.self
+@section("mysection") let metatype3: Any.Type = S.self
+@section("mysection") let metatype4: any (Hashable & Sendable).Type = Int.self
+@section("mysection") let metatype5: any (Hashable & Sendable).Type = S.self
```

Comment:
Can you add tests for bound generic types, e.g. `Array<Int>.self`?

Reply:
We're rejecting that case syntactically: https://github.com/swiftlang/swift/pull/85139/files#diff-ea7de05f211d676ea604c35f0e6b253fb4a454efbfcc84928a931bab113e7bdcR82